crates/nostr/src/filter.rs

```diff
@@ -364,6 +364,150 @@ pub struct Filter {
     pub generic_tags: GenericTags,
 }
 
+/// Specifies which event fields to compare with a filter.
+#[derive(
+    Debug, Clone, Default, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+pub struct MatchEventOptions {
+    /// Compare the event id with the filter.
+    pub id: bool,
+    /// Compare the event author with the filter.
+    pub author: bool,
+    /// Compare the event kind with the filter.
+    pub kind: bool,
+    /// Compare the event tags with the filter.
+    pub tags: bool,
+    /// Check if event timestamp is after filter's `since`.
+    pub since: bool,
+    /// Check if event timestamp is before filter's `until`.
+    pub until: bool,
+    /// Check if event content contains filter's search terms
+    /// (basic NIP-50 implementation without extensions).
+    pub nip50: bool,
+}
+
+impl MatchEventOptions {
+    /// Creates a new [`MatchEventOptions`] that matches all event fields by default.
+    #[inline]
+    pub fn new() -> Self {
+        Self {
+            id: true,
+            author: true,
+            kind: true,
+            tags: true,
+            since: true,
+            until: true,
+            nip50: true,
+        }
+    }
+
+    /// Returns a new [`MatchEventOptions`] that won't match any event.
+    #[inline]
+    pub fn unmatch() -> Self {
+        Self::default()
+    }
+
+    /// Compare the event id with the filter.
+    #[inline]
+    pub fn match_id(mut self) -> Self {
```

Comment:
What do you think if we use `<func-name>(mut self, enable: bool) -> Self` instead of useing `match_*` and `unmatch_*`? So in this case:

```rust
/// Compare the event id with the filter.
    #[inline]
    pub fn id(mut self, enable: bool) -> Self {
        self.id = enable;
        self
    }
```